Simplenote/Classes/SPSettingsViewController+Extensions.swift

```diff
@@ -88,3 +88,77 @@ private extension BiometricAuthentication.Biometry {
         }
     }
 }
+
+// MARK: - Account Deletion
+extension SPSettingsViewController {
+
+    @objc
+    func deleteAccountWasPressed() {
+        guard let user = SPAppDelegate.shared().simperium.user else {
+            return
+        }
+        SPTracker.trackDeleteAccountButttonTapped()
+        let pendingAlert = ActivityIndicatorAlertViewController()
+        pendingAlert.startAnimating()
+
+        let deletionController = AccountDeletionController()
+        deletionController.successHandler = {
+            pendingAlert.stopAnimating()
+            pendingAlert.dismiss(animated: false, completion: nil)
+            self.presentSuccessAlert(for: user)
+        }
+
+        deletionController.failureHandler = { status in
+            if status == 0 {
+                pendingAlert.stopAnimating()
+                pendingAlert.dismiss(animated: false, completion: nil)
+                NoticeController.shared.present(NoticeFactory.networkError())
+                return
+            }
+
+            let alert = UIAlertController(title: AccountDeletion.errorTitle, message: AccountDeletion.errorMessage, preferredStyle: .alert)
+            alert.addDefaultActionWithTitle(AccountDeletion.ok)
+        }
+
+        presentAccountDeletionConfirmation { (_) in
```

Comment:
I think it's a good idea to keep `onConfirm` here for clarity, otherwise it's not clear what the callback is.
And if we don't use `UIAlertAction`, let's remove it.